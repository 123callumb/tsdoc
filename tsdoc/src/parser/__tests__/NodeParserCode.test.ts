import { TestHelpers } from './TestHelpers';

test('00 Code span basic, positive', () => {
  TestHelpers.parseAndMatchNodeParserSnapshot([
    '/**',
    ' * line `1`',
    ' * line ` 2` sdf',
    ' */'
  ].join('\n'));
});

test('01 Code span basic, negative', () => {
  TestHelpers.parseAndMatchNodeParserSnapshot([
    '/**',
    ' * `multi',
    ' * line`',
    ' */'
  ].join('\n'));
  TestHelpers.parseAndMatchNodeParserSnapshot([
    '/**',
    ' * one`two',
    ' * `three`four',
    ' */'
  ].join('\n'));
});
