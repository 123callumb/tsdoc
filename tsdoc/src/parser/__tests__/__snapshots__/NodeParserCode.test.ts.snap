// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`00 Code span basic, positive 1`] = `
Object {
  "buffer": "/**[n] * line [c]1[c][n] * line [c] 2[c] sdf[n] */",
  "lines": Array [
    "line [c]1[c]",
    "line [c] 2[c] sdf",
  ],
  "rootNode": Object {
    "kind": "Section",
    "nodes": Array [
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "line ",
      },
      Object {
        "kind": "CodeSpan",
        "nodeExcerpt": "[c]1[c]",
      },
      Object {
        "kind": "SoftBreak",
        "nodeExcerpt": "[n]",
      },
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "line ",
      },
      Object {
        "kind": "CodeSpan",
        "nodeExcerpt": "[c] 2[c]",
      },
      Object {
        "kind": "PlainText",
        "nodeExcerpt": " sdf",
      },
      Object {
        "kind": "SoftBreak",
        "nodeExcerpt": "[n]",
      },
    ],
  },
}
`;

exports[`01 Code span basic, negative 1`] = `
Object {
  "buffer": "/**[n] * [c]multi[n] * line[c][n] */",
  "lines": Array [
    "[c]multi",
    "line[c]",
  ],
  "rootNode": Object {
    "kind": "Section",
    "nodes": Array [
      Object {
        "errorLocation": "[c]",
        "errorMessage": "The code span is missing its closing backtick",
        "kind": "ErrorText",
        "nodeExcerpt": "[c]",
      },
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "multi",
      },
      Object {
        "kind": "SoftBreak",
        "nodeExcerpt": "[n]",
      },
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "line",
      },
      Object {
        "errorLocation": "[c]",
        "errorLocationPrecedingToken": "line",
        "errorMessage": "The opening backtick for a code span must be preceded by whitespace",
        "kind": "ErrorText",
        "nodeExcerpt": "[c]",
      },
      Object {
        "kind": "SoftBreak",
        "nodeExcerpt": "[n]",
      },
    ],
  },
}
`;

exports[`01 Code span basic, negative 2`] = `
Object {
  "buffer": "/**[n] * one[c]two[n] * [c]three[c]four[n] */",
  "lines": Array [
    "one[c]two",
    "[c]three[c]four",
  ],
  "rootNode": Object {
    "kind": "Section",
    "nodes": Array [
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "one",
      },
      Object {
        "errorLocation": "[c]",
        "errorLocationPrecedingToken": "one",
        "errorMessage": "The opening backtick for a code span must be preceded by whitespace",
        "kind": "ErrorText",
        "nodeExcerpt": "[c]",
      },
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "two",
      },
      Object {
        "kind": "SoftBreak",
        "nodeExcerpt": "[n]",
      },
      Object {
        "errorLocation": "[c]",
        "errorLocationPrecedingToken": "three",
        "errorMessage": "Error parsing code span: The closing backtick for a code span must be followed by whitespace",
        "kind": "ErrorText",
        "nodeExcerpt": "[c]",
      },
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "three",
      },
      Object {
        "errorLocation": "[c]",
        "errorLocationPrecedingToken": "three",
        "errorMessage": "The opening backtick for a code span must be preceded by whitespace",
        "kind": "ErrorText",
        "nodeExcerpt": "[c]",
      },
      Object {
        "kind": "PlainText",
        "nodeExcerpt": "four",
      },
      Object {
        "kind": "SoftBreak",
        "nodeExcerpt": "[n]",
      },
    ],
  },
}
`;
